client/lib/purchases/index.js

```diff
@@ -45,7 +45,8 @@ function getPurchasesBySite( purchases, sites ) {
 				 * we fall back on the domain. */
 				slug: siteObject ? siteObject.slug : currentValue.domain,
 				title: currentValue.siteName || currentValue.domain || '',
-				purchases: [ currentValue ]
+				purchases: [ currentValue ],
+				url: siteObject.URL
```

Comment:
Yes, I think the change here should be to do `siteObject ? siteObject.domain : currentValue.domain` in the domain attribute.